tests/back/Platform/Specification/Component/EventQueue/BusinessEventNormalizerSpec.php

```diff
@@ -96,10 +121,45 @@ public function it_does_not_supports_denormalization_of_non_business_event(): vo
             ->shouldReturn(false);
     }
 
-    public function it_denormalizes_a_business_event()
+    public function it_denormalizes_a_business_event(UserInterface $user)
     {
-        $businessEvent = new class ('author', ['data'], 0, 'e0e4c95d-9646-40d7-be2b-d9b14fc0c6ba') extends BusinessEvent
-        {
+        $user->getUsername()->willReturn('julia');
+        $user->getFirstName()->willReturn('Julia');
+        $user->getLastName()->willReturn('Doe');
+        $user->isApiUser()->willReturn(true);
+
+        $author = Author::fromUser($user->getWrappedObject());
+
+        $businessEvent = new class ($author, ['data'], 0, 'e0e4c95d-9646-40d7-be2b-d9b14fc0c6ba') extends BusinessEvent {
+            public function name(): string
+            {
+                return 'event_name';
+            }
+        };
+
+        $data = [
+            'name' => 'event_name',
+            'author' => 'julia',
+            'author_type' => 'api',
+            'data' => ['data'],
+            'timestamp' => 0,
+            'uuid' => 'e0e4c95d-9646-40d7-be2b-d9b14fc0c6ba',
+        ];
+
+        $denormalizedBusinessEvent = $this->denormalize($data, get_class($businessEvent));
+
+        Assert::assertEquals($denormalizedBusinessEvent->getWrappedObject(), $businessEvent);
+    }
+
+    public function it_does_not_denormalize_a_business_event_because_author_type_is_not_good(UserInterface $user)
```

Comment:
Just a minor preference over Iznogoud
```suggestion
    public function it_does_not_denormalize_a_business_event_because_author_type_is_invalid(UserInterface $user)
```